Core/clim-core/gadgets.lisp

```diff
@@ -1853,24 +1844,39 @@ and must never be nil.")
                             (- middle slider-button-half-long-dim)))))))))
 |#
 
-
-(defmethod convert-value-to-position ((pane slider-pane))
-  (multiple-value-bind (x1 y1 x2 y2) (bounding-rectangle* (sheet-region pane))
-    (let ((x1 (+ x1 8.0)) ; replace this with some rectangle-inset transform or something
-          (y1 (+ y1 8.0)))
-      (multiple-value-bind (good-dim1 good-dim2)
-          (if (eq (gadget-orientation pane) :vertical)
-              ; vertical orientation
-              (values (+ y1 (ash slider-button-short-dim -1))
-                      (- y2 (ash slider-button-short-dim -1)))
-              ; horizontal orientation
-              (values (+ x1 (ash slider-button-short-dim -1))
-                      (- x2 (ash slider-button-short-dim -1))))
-        (+ good-dim1 (* (- good-dim2 good-dim1)
-                        (if (zerop (gadget-range pane))
-                            0.5
-                          (/ (- (gadget-value pane) (gadget-min-value pane))
-                             (gadget-range pane)))))))))
+(flet ((compute-dims (slider)
+         (multiple-value-bind (x1 y1 x2 y2) (bounding-rectangle* (sheet-region slider))
+           ;; Offset is the distance from the bounding region to the
+           ;; slider's "rail" and then some, so the knob doesn't go
+           ;; beyond the rail too much.
+           (let ((offset (+ (floor slider-button-short-dim 2) 4)))
+             (if (eq (gadget-orientation slider) :vertical)
+                 (values (+ y1 offset) (- y2 offset))
+                 (values (+ x1 offset) (- x2 offset))))))
+       (unlerp (v a b)
+         (if (= a b)
+             0.5
+             (/ (- v a) (- b a)))))
```

Comment:
please make it a top-level function. it is harder to navigate top-level flet's and non-toplevel defmethods.